pandas/util/_decorators.py

```diff
@@ -286,7 +298,12 @@ def deprecate_nonkeyword_arguments(
         The specific name of the function to show in the warning
         message. If None, then the Qualified name of the function
         is used.
+
+    klass : Warning, optional
```

Comment:
Can you add a test for this keyword, and a test to check that ``Pandas40DeprecationWarning == DeprecationWarning``?

(i.e. for something that would raise a ``Pandas40DeprecationWarning`` check it with ``tm.assert_produces_warning(DeprecationWarning``)

Reply:
Sure! 
I'd also love to to hear your thoughts about the whatsnew message. I'm not sure if the term "deprecation policy" is correct and whether we need to document this change at all